### PR TITLE
 Update Basix interface to allow variants

### PR DESCRIPTION
**Pull request by mscroggs**
This is coupled with FEniCS/dolfinx#1651 and FEniCS/basix#252.

### CODE REVIEW

File: ffcx/codegeneration/access.py [hunk 1]
```diff
@@ -261,7 +261,7 @@ def cell_vertices(self, e, mt, tabledata, num_points):
         ufl_scalar_element, = set(coordinate_element.sub_elements())
         assert ufl_scalar_element.family() in ("Lagrange", "Q", "S")
 
-        basix_scalar_element = create_element(ufl_scalar_element)
+        basix_scalar_element = create_element(ufl_scalar_element, "equispaced")
```
Comment (garth-wells) [inline ffcx/codegeneration/access.py:264]: Could we make the lattice type an enum? (defined in Basix)

File: ffcx/element_interface.py [hunk 2]
```diff
@@ -25,30 +25,37 @@
 }
 
 
-def create_element(ufl_element):
+def create_element(ufl_element, *args):
```
Comment (garth-wells) [inline ffcx/element_interface.py:28]: I've always had a dislike of `*args` for readability reasons. Is there a simple way around using it? If no, we need a detailed docstring.

File: ffcx/ir/elementtables.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def clamp_table_small_numbers(table,
 
 
 def get_ffcx_table_values(points, cell, integral_type, ufl_element, avg, entitytype,
-                          derivative_counts, flat_component):
+                          derivative_counts, flat_component, finite_element_args):
```
Comment (garth-wells) [inline ffcx/ir/elementtables.py:57]: Add some docs - not sure what `finite_element_args` is.

**Review by garth-wells — CHANGES_REQUESTED** [review]

File: ffcx/ir/elementtables.py (hunk 3 above)
Comment (mscroggs) [reply ffcx/ir/elementtables.py]: Thankfully, `finite_element_args` is no longer necessary so the code is now a little simpler

File: ffcx/codegeneration/ufc.h [hunk 4]
```diff
@@ -123,6 +123,19 @@ extern "C"
...
 
+    /// Return the Basix identifier of the family of the finite element function space
+    int basix_family;
+
+    /// Return the Basix identifier of the cell shape
+    int basix_cell;
+
+    /// Set to true if Basix's create_element function for this element
+    /// needs to be given a lattice type
+    bool needs_lattice_type;
```
Comment (garth-wells) [inline ffcx/codegeneration/ufc.h:134]: Have about removing this and using `lattice_type = -1` if the lattice isn't needed?

**Review by garth-wells — APPROVED** [review]

**Pull request merged**